Port to 1.11

Raycoms opened the pull request:
# Changes proposed in this pull request:
- backports the most recent changes to 1.11


Review please

nojo121 committed:
Revert "fixed lumberjack using axe now. (#685)"

This reverts commit ffff12f0cf66969e40d0509a2235ba60c3bf745b.

nojo121 committed:
fixed lumberjack using axe now. (#685)

Raycoms committed:
important hotfix. (#703)

Raycoms committed:
Hotifx/get colony crash (#705)

* important hotfix.

* fix the colony list

Raycoms committed:
fix mirror building with style change (#691)

marchermans committed:
Fix configuration values not being clamped properly. (#710)

Closes #709.

Signed-off-by: OrionOnline <oriondevelopment@outlook.com>

Raycoms committed:
Fix bug with placing supplyships in others colonies. (#688)

Raycoms committed:
feature/#658 lj anti stuck

Raycoms committed:
fix merge conflicts

Raycoms committed:
merge lumberjack slime tree support

Raycoms committed:
Revert "merge lumberjack slime tree support"

This reverts commit 27821dd03dc9d3af7760e78204e8177b9ce1a45b.

Raycoms committed:
merge warehouse overflow issue fix

Raycoms committed:
merge the fix some colony commands

Raycoms committed:
merge colony permission changes

Raycoms committed:
fix up 1.11 conflicts

Raycoms committed:
fix small merge conflicts

MinecoloniesSonar commented:
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L1284](https://github.com/Minecolonies/minecolonies/blob/7208bcd14a52bf112de0b84417badc27c346326f/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L1284): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

MinecoloniesSonar commented:
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L1284](https://github.com/Minecolonies/minecolonies/blob/4d61dd4ea3eee3c81d7346eb6e6166eac2e801b3/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L1284): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Raycoms committed:
hotfix problem with supplyship placement

MinecoloniesSonar commented:
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L1284](https://github.com/Minecolonies/minecolonies/blob/e48f6626f41a0954da682cb601fa30f332cbaa9e/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L1284): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Raycoms committed:
fix problem with lumberjack merge

MinecoloniesSonar commented:
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L1284](https://github.com/Minecolonies/minecolonies/blob/8f0ab568849458e756b33182dd77656181ef9efa/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L1284): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

Raycoms committed:
fix the problem with missing colonies (#728)

Merging this fast as a hotfix, I don't think this is the best way to fix this. but I will eventually get make a new PR when I have time.

Raycoms committed:
merge fix rtp death

Raycoms committed:
remove creator

Raycoms committed:
hotfix colony list

Raycoms committed:
merge

Raycoms committed:
fix commands merge

Raycoms committed:
Fix up 1.11 porting errors

MinecoloniesSonar commented:
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractSingleCommand.java#L116](https://github.com/Minecolonies/minecolonies/blob/8383d0bee699f925c9489c5a435a415389b633e4/src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java#L116): The Cyclomatic Complexity of this method "canCommandSenderUseCommand" is 14 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L1284](https://github.com/Minecolonies/minecolonies/blob/8383d0bee699f925c9489c5a435a415389b633e4/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L1284): Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1066)
1. ![MAJOR][MAJOR] [RecallCitizenMessage.java#L84](https://github.com/Minecolonies/minecolonies/blob/8383d0bee699f925c9489c5a435a415389b633e4/src/main/java/com/minecolonies/coremod/network/messages/RecallCitizenMessage.java#L84): Avoid if (x != y) ..; else ..; [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AConfusingTernary)

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

marvin-bitterlich committed:
Merge branch 'version/1.11' into port-to-1.11

MinecraftPlaye committed:
Feature/#18 Using Ore Dictionary in crafting recipes (#732)

* Ore Dict

Implemented ore dictionary to crafting recipes.

* Constants

Use now constants for often used ore dicts

Raycoms committed:
Fix construction tape problem with different styles

Add more style style
Add alternative style

Raycoms committed:
Add a fix for fakePlayers crashing server

Raycoms committed:
fix up problems with old versions

Raycoms committed:
fix up null pointer exception when breaking building.

Raycoms committed:
config from ingame

Raycoms committed:
dessert style, op check and change structure sequence

Raycoms committed:
fix javadoc
